### DISCUSSION TITLE
HowTo Guide: From Benbox to LaserGRBL (LX-NANO/Nano-328p, CH340, Benbox, DCCduino)

### DISCUSSION THREAD

**Discussion by vuchl** [arkypita/LaserGRBL#1213] [Show and tell]
Hello Folks!

I wanted to share my path to getting one of those cheaper Benbox engravers setup and running pretty well with LaserGRBL. I struggled first with firmware and connectivity issues but it seems it is all pretty much sorted out now.

# Reason
The reason behind going with LaserRGBL instead of the out-of-the-box Benbox 3.7.99 software is, that I get a wider range of supported file types. Namely `.nc` and `.svg`. SVGs are my go to format for non-picture images such as schemes, text etc. You know the deal, vectors are great.

Also it seems like the LaserGRBL software has some much more advanced rasterization, dithering and image transmutation capabilities.

# The Engraver
The engraver is a 450µm blue laser with 3000mW of output power.
![image](https://user-images.githubusercontent.com/4921147/102583033-23ba0400-4104-11eb-92db-b1b93e60ab6e.png)

The laser is driven by 3 stepper motors for a 2-axis operation and needs to be manually refocused if the distance to the material that is to be engraved changes. The laser has a bed of 500x600mm and therefore a usable area of approx. 400x500mm.
![image](https://user-images.githubusercontent.com/4921147/102583194-827f7d80-4104-11eb-9a99-33af22b11a60.png)

The chip that drives the power board, laser and motors is an Ardunio Nano. From what I found out it's a chinese replica of an original Ardunio as it is using the CH340G chip. (The chip can not be seen here as it is on the botom side of the Arduino).
![image](https://user-images.githubusercontent.com/4921147/102583420-f883e480-4104-11eb-9bda-8fe5c1d02f36.png)

# Installing Benbox and getting rid of it
Since the engraver came OOTB with the Benbox software I went with it and installed the CH340 driver and Benbox according to the instructions ([Youtube video for assembly and installation](https://youtu.be/fdgIeWwFBcM)
, [another video on assembly](https://youtu.be/7jj5ea1MvEo) and [A Google Drive link with all the stuff that was on the CD](
https://drive.google.com/open?id=1POhkqtVhc1BuYKE0_5hlvFC4vqjK427U)).

Benbox ran fine but it lacks some functionality.

So after some time figuring out Benbox and testing it's capabilities I deinstalled it and rebooted. Rebooting is quite critical since it seems that the software is not cleaning up well and **somehow the COM port that is assigned to the driver is not usable in LaserGRBL without a restart**.

# Flashing the GRBL 1.1h Homing XY firmware
Well before flashing you obviously need to install LaserGRBL. Did that. Straightforward. You can also skip installing Benbox and also install the CH340 driver from the LaserGRBL Tools menu. This one also seems to be newer (2019).

Next step to getting LaserGRBL working with the Benbox laser is to flash the GRBL firmware. I used the LaserGRBL integrated flashing tool and those settings:
![image](https://user-images.githubusercontent.com/4921147/102584164-303f5c00-4106-11eb-8bf2-a8dbefaf7713.png)
- Target: Arduino Nano
- Firmware: v1.1h custom, XY Homing [30.8.2019]
- Port: COM _x_ (see http://lasergrbl.com/usage/flash/ for more detailed overall flashing instructions)
- Baud Rate: 57600

After a succesful flash I was able to **connect to my now GRBL compatible engraver via a Baud Rate of 115200** (this seems to be important as well as all other settings do time out (consult the log at ? -> Open session log)
![image](https://user-images.githubusercontent.com/4921147/102584770-63ceb600-4107-11eb-8062-fe4a15aeb387.png)

The succesful connection is acknowledged in the status window with
```
Grbl 1.1h ['$' for help]
[MSG: '$H' | '$X' to unlock]
```

This is also documented here: https://lasergrbl.com/usage/arduino-connection/

# GRBL Settings for operation
After getting a succesful connection to the device I followed [this guide](http://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf) which was linked [here](https://lasergrbl.com/configuration/) and worked my way through it

## Complete Config file
After quite some testing I figured out this GRBL configuration for the engraver to work really nicely. it's quick, precise and cuts/engraves crisp:
```nc
$0=200 (Step pulse time)
$1=25 (Step idle delay)
$2=0 (Step pulse invert)
$3=3 (Step direction invert)
$4=0 (Invert step enable pin)
$5=0 (Invert limit pins)
$6=0 (Invert probe pin)
$10=3 (Status report options)
$11=0.010 (Junction deviation)
$12=0.002 (Arc tolerance)
$13=0 (Report in inches)
$20=0 (Soft limits enable)
$21=0 (Hard limits enable)
$22=1 (Homing cycle enable)
$23=0 (Homing direction invert)
$24=25.000 (Homing locate feed rate)
$25=4000.000 (Homing search seek rate)
$26=250 (Homing switch debounce delay)
$27=1.000 (Homing switch pull-off distance)
$30=255 (Maximum spindle speed)
$31=0 (Minimum spindle speed)
$32=1 (Laser-mode enable)
$100=82.000 (X-axis travel resolution)
$101=82.000 (Y-axis travel resolution)
$102=0.000 (Z-axis travel resolution)
$110=4000.000 (X-axis maximum rate)
$111=4000.000 (Y-axis maximum rate)
$112=0.000 (Z-axis maximum rate)
$120=164.000 (X-axis acceleration)
$121=164.000 (Y-axis acceleration)
$122=0.000 (Z-axis acceleration)
$130=550.000 (X-axis maximum travel)
$131=400.000 (Y-axis maximum travel)
$132=0.000 (Z-axis maximum travel)
```

The important settings in my opinion are:

## Laser Timings
```nc
$0=200 (Step pulse time)
$1=25 (Step idle delay)
```
Allthough the are the defauts they work well.

## Inversion of directions to align with jog arrows
```nc
$3=3 (Step direction invert)
```
It seems from wiring and setting up the device I may have put the controller at the wrong location. So per the PDF and this section
![image](https://user-images.githubusercontent.com/4921147/102585454-c2486400-4108-11eb-927d-2141e549e1a7.png) I managed to get to this setting to invert both X and Y to have the directions applied the right way.

## Stepping motor speeds
```nc
$25=4000.000 (Homing search seek rate)
...
$110=4000.000 (X-axis maximum rate)
$111=4000.000 (Y-axis maximum rate)
```

Allright this was a tricky one. The Benbox software has a default feedrate of 1000. I put is times 4 since all the prints I tried came out 4x the actual size. I meassured the engraved outputs and made the adjustmenst to the feedrate and the motor stepping resolution.

## Motor step resolution
```nc
$100=82.000 (X-axis travel resolution)
$101=82.000 (Y-axis travel resolution)
```

This is basically where I came up with the 4000 from above. Benbox comes with a pixels per mm setting of 320. With this setting when I jogged 1 step, it made way of ~4mm. So the best I came up with by messuring different lengths was the 82 above.

320 / 84 is about 4. So I upped the feedrate from 1000 to 4000.

## Motor acceleration
```nc
$120=164.000 (X-axis acceleration)
$121=164.000 (Y-axis acceleration)
```

Not much thought went into this. 164 just provides a snappy and precise feeling when operating the engraver.

## Laser power defaults
```nc
$30=255 (Maximum spindle speed)
$31=0 (Minimum spindle speed)
```

This is the default I believe but works well

## PWM support aka "grey"
```nc
$32=1 (Laser-mode enable)
```

With this setting enabled the laser can modulate the power. Instead of only being on and off it has 256 different intensities which LaserGRBL can use if you setup the rasterization with `M4` mode. The setting is mandatory for being able to use `M4` at all. More on this [here](https://lasergrbl.com/usage/raster-image-import/target-image-size-and-laser-options/).

## Bed dimensions
```nc
$130=550.000 (X-axis maximum travel)
$131=400.000 (Y-axis maximum travel)
```
Setting the inner usable area of the board. The engraver has no limit switches so only soft limits apply.

# Known issues

I found that th `$H` command is not working and freezes the laser. So don't use the homing button for now.
![image](https://user-images.githubusercontent.com/4921147/102586884-35eb7080-410b-11eb-93c6-026ecc19ab02.png)

The jog Home button works "per session" as in: "when you start the laser and it uses the current position as it zero point it can get back there". But it can not locate the true zero point when it is powered off in between.

Also sometimes I need to reset the Arduino to properly function again. I just press the tiny white switch on top of the board then and I can start engraving again. Mind you: this is not keeping current progress of engraving.
![image](https://user-images.githubusercontent.com/4921147/102587944-f2920180-410c-11eb-92fd-48a68ab15d29.png)

Anyay, thanks @arkypita for building LaserGRBL. It's a really nice tool for free and let's starters have a quality engraving stack!

Regards Sebastian

**Comment by arkypita** [comment]
Fantastic! This is a wonderful contribution of know-how that you wanted to share in full community perspective.
I hope that google indexes it well and allows those looking for a guide for this procedure to find it.

About the Homing known issue: just disable homing feature in grbl configuration ($22 = 0) to get rid-off the homing button.
If your hardware does not have limit switches both homing and hard limits.

![image](https://user-images.githubusercontent.com/8782035/102605590-c20a9180-4125-11eb-89c6-3377e171874f.png)

**Comment by vuchl** [comment]
Thanks man! My next step is to provide entries for the material database for the nano based on the findings. But as a true beginner I need some experience ande right now in the lockdown it's not that easy to get a lot of differnt materials without online orders.

~~Also can you point me to the right direction for application translations. I think of doing the german translation for LaserGRBL.~~ Ther german translation is actually quite good. Maybe I put in a pull request for some of the missing ones,
